hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/adapters/IncompatibleMutationAdapter.java

```diff
@@ -47,6 +48,11 @@
   private final Connection connection;
   private final Configuration conf;
   private final MetricsExporter metricsExporter;
+  // Maximum timestamp (in usecs) that bigtable can handle while preserving lossless conversion to
+  // hbase timestamps in ms.
+  static final long BIGTABLE_MAX_TIMESTAMP = Long.MAX_VALUE - (Long.MAX_VALUE % 1000L);
```

Comment:
I don't understand this logic? max timestamp in usec that BT can support is LONG.MAX_VALUE?

Reply:
I was looking from hbase perspective. HBASE_EFFECTIVE_MAX_TIMESTAMP is the maximum timestamp which bigtable can accept without overflow. If the timestamp > HBASE_EFFECTIVE_MAX_TIMESTAMP, there will be overflow.

Reply:
so that MAX timestamp bigtable can handle is `Long.MAX_VALUE / 1000` I don' t understand where `%` comes into picture.

Reply:
Replaced with it. this was just pruning the last 3 digits and doing the division.